sklearn/gaussian_process/gpc.py

```diff
@@ -0,0 +1,194 @@
+"""Gaussian processes classification based on the Laplace approximation. """
+
+# Authors: Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+#
+# License: BSD 3 clause
+
+import numpy as np
+from scipy.linalg import cholesky, cho_solve, solve
+from scipy.optimize import fmin_l_bfgs_b
+from scipy.special import erf
+
+from sklearn.base import BaseEstimator
+from sklearn.gaussian_process.kernels import RBF
+from sklearn.utils.validation import check_X_y, check_is_fitted, check_array
+
+
+# Values required for approximating the logistic sigmoid by
+# error functions. coefs are obtained via:
+# x = np.array([0, 0.6, 2, 3.5, 4.5, np.inf])
+# b = logistic(x)
+# A = (erf(np.dot(x, self.lambdas)) + 1) / 2
+# coefs = lstsq(A, b)[0]
+LAMBDAS = np.array([0.41, 0.4, 0.37, 0.44, 0.39])[:, np.newaxis]
+COEFS = np.array([-1854.8214151, 3516.89893646, 221.29346712,
+                  128.12323805, -2010.49422654])[:, np.newaxis]
+
+
+class GaussianProcessClassifier(BaseEstimator):
+    """ Gaussian process classification (GPC).
+
+    The implementation is based on Algorithm 3.1, 3.2, and 5.1 of
+    ``Gaussian Processes for Machine Learning'' (GPML) by Rasmussen and
+    Williams.
+
+    Internally, the Laplace approximation is used for approximating the
+    non-Gaussian posterior by a Gaussian.
+
+    Currently, the implementation is restricted to
+      * using the logistic link function
+      * binary classification
+    """
+
+    def __init__(self, kernel=RBF(), jitter=0.0, optimizer="fmin_l_bfgs_b"):
```

Comment:
initializing with `RBF()` is a bad idea, since this is a unique object that is mutable, so it may undergo unexpected changes during execution of a program (same reason never to put `default=[]` or `default={}`)